src/datasets/arrow_dataset.py

```diff
@@ -1384,6 +1385,44 @@ def with_transform(
         dataset.set_transform(transform=transform, columns=columns, output_all_columns=output_all_columns)
         return dataset
 
+    def prepare_for_task(self, task: Union[str, TaskTemplate]) -> "Dataset":
+        """Prepare a dataset for the given task.
```

Comment:
I think it's good to explain a bit what are the operations contained in the expression "prepare a dataset"

Reply:
good idea! i opted for the following:

> Prepare a dataset for the given task by casting the dataset's :class:`Features` to standardized column names and types as detailed in `~tasks`.